### PR TITLE
FEATURE: SD-1292 Adding styles for submenu component

### PR DESCRIPTION
**Pull request by thuytran-gooddata**
Changing styles of new component and we don't change logic code. So we dont need Code owners (Rail team) to review this PR.

---
Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1981/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3612/ : FAILURE in 5m 37s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1982/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3613/ : FAILURE in 5m 32s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1983/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3614/ : FAILURE in 7m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1984/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3615/ : FAILURE in 6m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1985/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3616/ : SUCCESS in 14m 45s

**Comment by thuytran-gooddata** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1986/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/580/ : SUCCESS in 20m 42s

**Comment by thuytran-gooddata** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1987/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3617/ : SUCCESS in 15m 02s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1988/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/581/ : SUCCESS in 21m 01s

**Comment by anvuong119** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1989/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3618/ : SUCCESS in 14m 51s

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 1]
```diff
@@ -602,3 +607,94 @@ $gd-header-project-width: 230px;
...
+.gd-data-header-menu-section {
+    display: flex;
+    flex: 1 0 auto;
+
+    .icon-sources {
+        border-bottom-right-radius: 0;
+        border-top-right-radius: 0;
+
+        &:hover {
+            border-color: #27b3e0;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:620]: should we define a variable to store the value of `border-color` and then reuse the variable for lines 628 and 642 below

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 2]
```diff
@@ -602,3 +607,94 @@ $gd-header-project-width: 230px;
...
+        }
+    }
+
+    .icon-model {
+        border-radius: 0;
+
+        &:hover {
+            border-color: #27b3e0;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:628]: we can reuse the variable defined previously here.

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 3]
```diff
@@ -602,3 +607,94 @@ $gd-header-project-width: 230px;
...
+        border-bottom-left-radius: 0;
+        border-top-left-radius: 0;
+
+        &::before {
+            top: 11px;
+            color: #b0beca;
+        }
+
+        &:hover {
+            border-color: #27b3e0;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:642]: we can reuse the variable defined previously here.

**Comment by anvuong119** [comment]
Please tick the checkboxes in the `PR Checklist` above. Also, please don't forget to enter a comment telling the reason why it's not necessary to ask the Code owners to review this PR.

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 1 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have defined and reuse it thanks

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 2 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have defined and reuse it thanks

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 3 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have defined and reuse it thanks

**Comment by thuytran-gooddata** [comment]
> Please tick the checkboxes in the `PR Checklist` above. Also, please don't forget to enter a comment telling the reason why it's not necessary to ask the Code owners to review this PR.
I have updated the comment and checklist thanks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1990/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3619/ : SUCCESS in 14m 11s

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 4]
```diff
@@ -602,3 +608,94 @@ $gd-header-project-width: 230px;
...
+.gd-data-header-menu-section {
+    display: flex;
+    flex: 1 0 auto;
+
+    .icon-sources {
+        border-bottom-right-radius: 0;
+        border-top-right-radius: 0;
+
+        &:hover {
+            border-color: $button-border-hover;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:621]: why don't we use `$gd-color-highlight` here to make it consistent with the styling of Gooddata

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 5]
```diff
@@ -602,3 +608,94 @@ $gd-header-project-width: 230px;
...
+        }
+    }
+
+    .icon-model {
+        border-radius: 0;
+
+        &:hover {
+            border-color: $button-border-hover;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:629]: why don't we use `$gd-color-highlight` here to make it consistent with the styling of Gooddata

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 6]
```diff
@@ -602,3 +608,94 @@ $gd-header-project-width: 230px;
...
+        border-bottom-left-radius: 0;
+        border-top-left-radius: 0;
+
+        &::before {
+            top: 11px;
+            color: #b0beca;
+        }
+
+        &:hover {
+            border-color: $button-border-hover;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:643]: why don't we use `$gd-color-highlight` here to make it consistent with the styling of Gooddata

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 7]
```diff
@@ -9,6 +9,12 @@ $gd-header-height: 44px;
 $gd-header-animation-easing: cubic-bezier(0.23, 1, 0.32, 1);
 $gd-header-project-width: 230px;
 
+$button-border-hover: #27b3e0;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:12]: If this variable can be replaced by `$gd-color-highlight`, then I think it's not necessary anymore. If that's right, please remove the variable.

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 8]
```diff
@@ -602,3 +608,94 @@ $gd-header-project-width: 230px;
...
+        }
+    }
+
+    .icon-cloud-upload {
+        border-bottom-left-radius: 0;
+        border-top-left-radius: 0;
+
+        &::before {
+            top: 11px;
+            color: #b0beca;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:639]: should we use `#6d7680` as the color here, as used in the mockup?

File: libs/sdk-ui-kit/styles/scss/header.scss [hunk 9]
```diff
@@ -602,3 +608,94 @@ $gd-header-project-width: 230px;
...
+    button.is-active[class*="gd-button-"]::before,
+    button[class*="gd-button-"]::before {
+        width: 19px;
+        height: 16px;
+        font-size: 16px;
+    }
+
+    .gd-button-primary[class*="icon-"] {
+        box-shadow: none;
+        padding-left: 35px;
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/header.scss:656]: Is this the padding between the upload icon and the left edge of the button? If that's right, I think the value should be `11px` as used in the mockup.

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 4 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have replaced with it thanks

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 5 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have replaced with it thanks

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 6 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: I have replaced with it thanks

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 7 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: thank you we can use this for consistent with the styling of Gooddata. I just pick the color from modeler app to consistent wigth it but I think we should use gd-color-highlight and update in modeler later

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 8 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: changed it as mockup styling

File: libs/sdk-ui-kit/styles/scss/header.scss (hunk 9 above)
Comment (thuytran-gooddata) [reply libs/sdk-ui-kit/styles/scss/header.scss]: Changed it thanks

**Comment by anvuong119** [comment]
Please, tick all the checkboxes in `PR Checklist` and add a comment explaining the reason why it's not necessary to ask Code owners (Rail team) to review this PR.

**Comment by thuytran-gooddata** [comment]
> Please, tick all the checkboxes in `PR Checklist` and add a comment explaining the reason why it's not necessary to ask Code owners (Rail team) to review this PR.

I have updated it thanks

**Comment by anvuong119** [comment]
LGTM

**Review by lvkhanh — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/638/ : SUCCESS in 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/899/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3620/ : SUCCESS in 14m 19s

**Pull request merged**